File: examples/chapter12/advdiffeig.m
%%
% The eigenvalues of advection diffusion are near-imaginary for
% $\epsilon\approx 0$ and more negative-real for increasing values of
% $\epsilon$.
[x,Dx,Dxx] = diffper(40,[0,1]);
tau = 0.1;
for ep = [0.001 0.01 0.05]
  lambda = eig(-Dx + ep*Dxx);
  plot(real(tau*lambda),imag(tau*lambda),'.'), hold on
end
axis equal,  grid on    % ignore this line
legend('\epsilon = 0.001','\epsilon = 0.01','\epsilon = 0.05','location','northwest')    % ignore this line
title('Eigenvalues for advection-diffusion')    % ignore this line

File: julia/scripts/chapter12/advdiffeig.jl

include("../FNC.jl")
x,Dx,Dxx = FNC.diffper(40,[0,1]);

using LinearAlgebra,Plots
tau = 0.1
for ep = [0.001 0.01 0.05]
  lambda = eigvals(-Dx + ep*Dxx)
  scatter!(real(tau*lambda),imag(tau*lambda),m=3,label="\\epsilon=$ep")
end
plot!([],[],label="",title="Eigenvalues for advection-diffusion",leg=:left,aspect_ratio=1)     
